src/server.c

```diff
@@ -5911,26 +5911,55 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
         }
     }
 
+    /* Threads */
+    int stat_io_ops_processed_calculated = 0;
+    long long stat_io_reads_processed = 0, stat_io_writes_processed = 0;
+    long long stat_total_reads_processed = 0, stat_total_writes_processed = 0;
+    if (all_sections || (dictFind(section_dict,"threads") != NULL)) {
+        if (sections++) info = sdscat(info,"\r\n");
+        info = sdscatprintf(info, "# Threads\r\n");
+        long long reads, writes;
+        for (j = 0; j < server.io_threads_num; j++) {
+            atomicGet(server.stat_io_reads_processed[j], reads);
+            atomicGet(server.stat_io_writes_processed[j], writes);
+            info = sdscatprintf(info, "io_thread_%d:clients=%d,reads=%lld,writes=%lld\r\n",
+                                       j, server.io_threads_clients_num[j], reads, writes);
+            stat_total_reads_processed += reads;
+            if (j != 0) stat_io_reads_processed += reads; /* Skip the main thread */
```

Comment:
shouldn't `stat_io_reads_processed` include the main thread?

Reply:
Good point. The variable `stat_io_reads_processed` is to show `io_threaded_reads_processed` metrics. Before async io thread feature, we update it when redis uses IO threads and main thread to process client requests in `beforeSleep`, but doesn't update it if not using IO threads to handle even threaded IO is enabled. So i think this metric is to show the workload of io threads(of course including main thread before).
But async io thread is different now, If `io_threaded_reads_processed` includes the main thread's now, `io_threaded_reads_processed` is always the same as `total_reads_processed ` if enabling threaded IO, it will be useless.

Reply:
i think the intention of this metric has not changed, but the data has changed somewhat. @oranagra could you please take a look?